src/structs/lepton_decoder.rs

```diff
@@ -289,30 +289,49 @@ fn parse_token<R: Read, const ALL_PRESENT: bool>(
 ) -> Result<()> {
     debug_assert!(pt.is_all_present() == ALL_PRESENT);
 
+    let neighbors =
+        context.get_neighbor_data::<ALL_PRESENT>(image_data, context, num_non_zeros, pt);
```

Comment:
What about renaming `num_non_zeros`? It is not the only and not even the main component of `NeighborSummary`.

Reply:
Also `context` is already there, should not send it inside the function again.